safehttp/response_writer.go

```diff
@@ -53,21 +57,54 @@ func NotWritten() Result {
 	return Result{}
 }
 
+// commitPhase calls the Commit phases of all the interceptors. This stage will
+// run before a response is written to the ResponseWriter. If a response is
+// written to the ResponseWriter in a Commit phase then the Commit phases of the
+// remaining interceptors won't execute.
+//
+// TODO: BIG WARNING, if ResponseWriter.Write and ResponseWriter.WriteTemplate
+// are called in Commit then this will recurse. CommitResponseWriter was an
```

Comment:
What happened to CommitResponseWriter? Don’t leave the reader hanging :)

Reply:
@kele suggested to leave this feature (and other possible restrictions to the `Commit` phase that we haven't foreseen) for another PR. :)